### PR TITLE
router: init at 1.14.0

### PR DESCRIPTION
**Pull request by bbigras**
###### Description of changes

https://github.com/apollographql/router

I stole the 'librusty_v8' part from pkgs/development/web/deno

Is `licenses.elastic` right for Elastic License 2?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/router/librusty_v8.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# auto-generated file -- DO NOT EDIT!
```
Comment (SuperSandro2000) [inline pkgs/servers/http/router/librusty_v8.nix:1]: How is that auto generated?

File: pkgs/servers/http/router/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, librusty_v8 ? callPackage ./librusty_v8.nix { }
```
Comment (SuperSandro2000) [inline pkgs/servers/http/router/default.nix:6]: If the version is hardcoded anyway, we don't need it as an input.

```suggestion
```

File: pkgs/servers/http/router/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    protobuf
+  ];
+
+  buildInputs = [
+    elfutils
+  ];
+
+  # The v8 package will try to download a `librusty_v8.a` release at build time to our read-only filesystem
+  # To avoid this we pre-download the file and export it via RUSTY_V8_ARCHIVE
+  RUSTY_V8_ARCHIVE = librusty_v8;
```
Comment (SuperSandro2000) [inline pkgs/servers/http/router/default.nix:35]:
```suggestion
  RUSTY_V8_ARCHIVE = callPackage ./librusty_v8.nix { };
```

File: pkgs/servers/http/router/librusty_v8.nix (hunk 1 above)
Comment (bbigras) [reply pkgs/servers/http/router/librusty_v8.nix:1]: No idea. That was in pkgs/development/web/deno/librusty_v8.nix. I'll remove the comment.

Comment (SuperSandro2000) [reply pkgs/servers/http/router/librusty_v8.nix]: I think it is generated from their update script

**Pull request merged**